### PR TITLE
Playlist and filters

### PR DESCRIPTION
**Pull request by bhushannaikaroskar**
Hey folks please review my PR for Playlist and Filters. 
(Note: I have not added some categories of videos so some filters wont display any results but they are working)

Live link: https://6284077171221a0d2e1527bc--boisterous-daffodil-390614.netlify.app/

Screenshot 2022-05-18 023857

Screenshot 2022-05-18 024536

### CODE REVIEW

**Review by developer-anand — APPROVED** [review]
Good Work, code is readable and naming conventions are also good.

File: src/components/playlistpage/SelectPlaylistModal.jsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+                                    checked={obj.videos.find(
+                                        (vid) => vid._id === video._id
+                                    )}
+                                    onClick={() => {togglePlaylistHandler(obj,video)}}
+                                />
+                                <label
+                                    className="playlist-checkbox-label"
+                                    htmlFor={obj._id}
+                                >
+                                    {" "}
```
Comment (iRohitGaur) [inline src/components/playlistpage/SelectPlaylistModal.jsx:47]: suggestion: might not be needed. check once

File: src/context/PlaylistProvider.jsx [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+        }
+        axios
+            .request({
+                method: "get",
+                url: "/api/user/playlists",
+                headers: { authorization: auth.authToken },
+                data: {},
+            })
+            .then((res) => {
+                console.log(res.data.playlists);
```
Comment (iRohitGaur) [inline src/context/PlaylistProvider.jsx:26]: suggestion: remove `console.log`

File: src/context/PlaylistProvider.jsx [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+                setPlaylist([...res.data.playlists]);
+            })
+            .catch((err) => {
+                console.log(err);
+            });
+    };
+
+    const addPlaylist = (title, description) => {
+        if (!auth.isAuthenticated) {
+            console.log("User is not Authenticated");
```
Comment (iRohitGaur) [inline src/context/PlaylistProvider.jsx:36]: suggestion: remove `console.log`

File: src/context/PlaylistProvider.jsx [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+                headers: { authorization: auth.authToken },
+                data: {
+                    playlist: {
+                        title,
+                        description,
+                    },
+                },
+            })
+            .then((res) => {
+                console.log(res.data.playlists);
```
Comment (iRohitGaur) [inline src/context/PlaylistProvider.jsx:52]: suggestion: remove `console.log`

File: src/context/PlaylistProvider.jsx [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+                setPlaylist([...res.data.playlists]);
+            })
+            .catch((err) => {
+                console.log(err);
+            });
+    };
+
+    const removePlaylist = (playlistId) => {
+        if (!auth.isAuthenticated) {
+            console.log("User is not Authenticated");
```
Comment (iRohitGaur) [inline src/context/PlaylistProvider.jsx:62]: suggestion: remove `console.log`

File: src/context/PlaylistProvider.jsx [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+        }
+        axios
+            .request({
+                method: "delete",
+                url: `/api/user/playlists/${playlistId}`,
+                headers: { authorization: auth.authToken },
+                data: {},
+            })
+            .then((res) => {
+                console.log(res.data.playlists);
```
Comment (iRohitGaur) [inline src/context/PlaylistProvider.jsx:73]: suggestion: remove `console.log`

File: src/context/PlaylistProvider.jsx [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+                setPlaylist([...res.data.playlists]);
+            })
+            .catch((err) => {
+                console.log(err);
+            });
+    };
+
+    const addToPlaylist = (playlistId, video) => {
+        if (!auth.isAuthenticated) {
+            console.log("User is not Authenticated");
```
Comment (iRohitGaur) [inline src/context/PlaylistProvider.jsx:83]: suggestion: remove `console.log`

File: src/context/PlaylistProvider.jsx [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+            .request({
+                method: "post",
+                url: `/api/user/playlists/${playlistId}`,
+                headers: { authorization: auth.authToken },
+                data: {
+                    video,
+                },
+            })
+            .then((res) => {
+                console.log(res.data.playlist);
```
Comment (iRohitGaur) [inline src/context/PlaylistProvider.jsx:96]: suggestion: remove `console.log`

File: src/context/PlaylistProvider.jsx [hunk 9]
```diff
@@ -0,0 +1,151 @@
...
+                ]);
+            })
+            .catch((err) => {
+                console.log(err);
+            });
+    };
+
+    const removeFromPlaylist = (playlistId, video) => {
+        if (!auth.isAuthenticated) {
+            console.log("User is not Authenticated");
```
Comment (iRohitGaur) [inline src/context/PlaylistProvider.jsx:112]: suggestion: remove `console.log`

File: src/context/PlaylistProvider.jsx [hunk 10]
```diff
@@ -0,0 +1,151 @@
...
+        axios
+            .request({
+                method: "delete",
+                url: `/api/user/playlists/${playlistId}/${video._id}`,
+                headers: { authorization: auth.authToken },
+                data: {
+                },
+            })
+            .then((res) => {
+                console.log(res.data.playlist);
```
Comment (iRohitGaur) [inline src/context/PlaylistProvider.jsx:124]: suggestion: remove `console.log`

**Review by iRohitGaur — APPROVED** [review]
Nice work!

Gave some suggestions, please look into it.

Good to merge with updates 👍🏻

File: src/context/PlaylistProvider.jsx (hunk 10 above)
Comment (bhushannaikaroskar) [reply src/context/PlaylistProvider.jsx:124]: The console.log files are actually error that I will convert to toasts. So kept it on purpose as reminder while using toasts

File: src/components/playlistpage/SelectPlaylistModal.jsx (hunk 1 above)
Comment (bhushannaikaroskar) [reply src/components/playlistpage/SelectPlaylistModal.jsx:47]: Didn't get you. Do you mean the entire component might be not needed or that string brackets. If you saying for those brackets those are added on refactor automatically will change that

**Pull request merged**